File: iOS - Using Objective-C/day1/day1Task5_ValidateNumberOrText/day1Task5_ValidateNumberOrText/ViewController.m
//
//  ViewController.m
//  day1Task5_ValidateNumberOrText
//
//  Created by JETSMobileLabMini5 on 17/12/2024.
//

#import "ViewController.h"
#define IS_TEXT  0
#define IS_NUMBER 1
@interface ViewController ()
@property (weak, nonatomic) IBOutlet UILabel *status;
@property (weak, nonatomic) IBOutlet UITextField *inputField;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}
- (IBAction)checkStatus:(id)sender {
    NSNumberFormatter* numberFormatter = [[NSNumberFormatter alloc] init] ;
    NSNumber* number = [numberFormatter numberFromString:_inputField.text];
    bool isNumber = false;
    if(number != nil){
        isNumber = true;
    }
    if ([_inputField.text isEqual:@""]) {
        _status.text = @"No input, enter data";
    }else{
        switch([sender tag]){
            case IS_TEXT:
                if(!isNumber){
                    _status.text = @"yes, it is text";
                }else{
                    _status.text = @"No, it is not number";
                }
                break;
            case IS_NUMBER:
                if(isNumber){
                    _status.text = @"yes, it is number";
                }else{
                    _status.text = @"No, it is not number";
                }
                break;
        }
    }
}


@end

File: Design Pattern/Day2/lec2Demo/lec2Demo/Controller/ViewController.swift
//
//  ViewController.swift
//  lec2Demo
//
//  Created by Usef on 23/01/2025.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var myLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        var products : [Product]?
        let service : NetworkProtocol = Service()
        service.fetchDataFromJson { [weak self] res in
            products = (res?.products)!
            DispatchQueue.main.async {
                self?.myLabel.text = products![0].title
            }
        }
    }


}

